### PR TITLE
ENH: Make `np.linalg.pinv` behave like a gufunc

### PR DESCRIPTION
**Pull request by jaimefrio**
Use the gufunc functionality of SVD to allow solving for multiple
pseudo-inverses with a single function call.

### CODE REVIEW

**Comment by charris** [comment]
Should be possible to generate test cases by using a matrix with normally distributed elements and then checking the results by 1) are the orthogonal matrices orthogonal and 2) does the result reproduce the input when multiplied together. We cannot yet do stacked matrix multiplication, but einsum should work for test purposes.

Needs rebase.

**Comment by eric-wieser** [comment]
This duplicates #8827. @jaimefrio, is this one a better call? If so, can you rebase?

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -1583,15 +1586,16 @@ def pinv(a, rcond=1e-15 ):
...
-            s[i] = 1./s[i]
-        else:
-            s[i] = 0.;
-    res = dot(transpose(vt), multiply(s[:, newaxis], transpose(u)))
+    m = u.shape[-2]
+    n = vt.shape[-1]
+    cutoff = rcond * maximum.reduce(s, axis=-1, keepdims=True)
+    mask = s > cutoff
+    s[mask] = 1. / s[mask]
+    s[~mask] = 0
```
Comment (eric-wieser) [inline numpy/linalg/linalg.py]: How does the performance here compare to `np.where(mask, 1/s, 0)`?

Comment (eric-wieser) [inline numpy/linalg/linalg.py]: Nevermind, `np.divide(1, s, where=mask, out=s)` is better

Comment (MSeifert04) [inline numpy/linalg/linalg.py]: Interesting, the `where` argument isn't even documented.

Comment (eric-wieser) [inline numpy/linalg/linalg.py]: The arguments are documented on the [ufuncs](https://docs.scipy.org/doc/numpy/reference/ufuncs.html#optional-keyword-arguments) page. I'd argue that we should remove the documentation of the `out` argument from all the ufuncs, and just include a reference back to the full documentation.

**Pull request closed without merging**